### PR TITLE
Transition ``pyreverse`` from ``optparse`` to ``argparse``

### LINKED ISSUE #6279: Migrate ``pyreverse`` to ``argparse``
### Current problem

Opening a separate issue to track progress here.

### Desired solution

After #6209 has landed I have a PR ready that works on moving `pyreverse` to `argparse`.

After that transition we can add a number of deprecation warnings on unused classes. For example, `OptionsProviderMixIn`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Closes #6279

### CODE REVIEW

File: pylint/pyreverse/main.py [hunk 1]
```diff
@@ -2,17 +2,18 @@
 # For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
 
-"""%prog [options] <packages>.
+"""{prog} [options] <packages>.
 
   create UML diagrams for classes and modules in <packages>
 """
```
Comment (Pierre-Sassoulas) [inline pylint/pyreverse/main.py:8]:
```suggestion
"""Create UML diagrams for classes and modules."""
```
I think argparse is generating this text.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
LGTM !

File: pylint/pyreverse/main.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/pyreverse/main.py:8]: No we actually pass this as `usage`. That's why we need to change the `prog` as it using an old way of formatting.

Comment (Pierre-Sassoulas) [reply pylint/pyreverse/main.py:8]: Could we pass this as doc instead ?

Comment (DanielNoord) [reply pylint/pyreverse/main.py:8]: Not sure you what you mean with `as doc`? See https://docs.python.org/3/library/argparse.html

Comment (Pierre-Sassoulas) [reply pylint/pyreverse/main.py:8]: Sorry I meant ``description``, thank you for linking the documentation.

Comment (Pierre-Sassoulas) [reply pylint/pyreverse/main.py:8]: The automatically generated doc is vastly superior to what we can do:
```
python prog.py -h
usage: prog.py [-h] [--sum] N [N ...]
```

Comment (DanielNoord) [reply pylint/pyreverse/main.py:8]: Fine with me! I'll change it around!

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by DudeNr33 — APPROVED** [review]
Looks good to me!

**Pull request merged**